File: python/pangram/pangram.py
def is_pangram(sentence):
    """
    Check if a sentence is a pangram (contains all 26 letters of the English alphabet).

    Args:
        sentence (str): The sentence to be checked.

    Returns:
        bool: True if the sentence is a pangram, False otherwise.
    """
    # Create a set to keep track of the unique letters in the sentence (case-insensitive)
    unique_letters = set()

    # Iterate through the sentence and add each letter to the set
    for char in sentence:
        if char.isalpha():
            unique_letters.add(char.lower())  # Convert to lowercase for case-insensitivity

    # Check if the set contains all 26 letters of the alphabet
    return len(unique_letters) == 26

File: javascript/pangram/pangram.js
//
// This is only a SKELETON file for the 'Pangram' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isPangram = (sentence) => {
  // Create a set to keep track of encountered letters
  const letterSet = new Set();

  // Convert the sentence to lowercase for case insensitivity
  const lowerCaseSentence = sentence.toLowerCase();

  // Iterate through each character in the sentence
  for (let char in lowerCaseSentence) {
    // Check if the character is a letter (a to z)
    if (/[a-z]/.test(char)) {
      letterSet.add(char); // Add the letter to the set 
    }
  }

  // Check if all 26 letters have been encountered
  return letterSet.size === 26;
};
